### PR TITLE
Issue 49: fixed getCA issue

### PR DESCRIPTION
**Pull request by tscherf**
This patch fixes the issue described in #55 and #49. Please review and merge when applicable.

### CODE REVIEW

**Comment by tedescn** [comment]
Are there any updates on the review or merging of this pull request?

Regards
Nigel

File: sscep.c [hunk 1]
```diff
@@ -619,7 +619,7 @@ main(int argc, char **argv) {
 			printf("%s: CA certificate written as %s\n",
 				pname, c_char);
 			(void)fclose(fp);
-			pkistatus = SCEP_PKISTATUS_SUCCESS;
```
Comment (mbartosch) [inline sscep.c]: I'm currently unable to test this patch, it looks fine overall, but I am a bit unsure about this line.
Doesn't this result in a missing status return code?

**Comment by mbartosch** [comment]
Sorry for forgetting to action on this. Please see above code comment. I am a bit reluctant to merge this without understanding the change for status propagation.
Unfortunately I don't have the time to test the submission myself, so I'd appreciate feedback if the patch actually does work properly.

**Comment by tedescn** [comment]
Martin,



Can you please confirm receipt or not of my initial response this morning? My fear is it didn’t appear within the pull request response.



Regards

Nigel Tedeschi



Security Trust and Mitigation, End User Technology & Security



BT Technology, Service & Operations |Tel:+44 (0)1473 608386 | Nigel.Tedeschi@BT.com<mailto:Nigel.Tedeschi@BT.com>

BT Group plc Registered office: 81 Newgate Street London EC1A 7AJ. Registered in England and Wales no. 4190816 This electronic message contains information from BT Group plc which may be privileged or confidential.  The information is intended to be for the use of the individual(s) or entity named above.  If you are not the intended recipient be aware that any disclosure, copying, distribution or use of the contents of this information is prohibited.  If you have received this electronic message in error, please delete it and notify me immediately by telephone or email.

**Comment by tedescn** [comment]
Martin,

I suspect the the second response appearing answers my assumption. The first email was:

Martin,

Good catch, I suspect there was an upload wrinkle where the replacement line has been lost during the upload activity? See attached original patch request that works for me.

FYI I also have a simple Microsoft batch script that exercises SCEP messages using the SSCEP client within a Microsoft Windows environment if you want it? I use this script to both demonstrate the protocol and  debug SCEP client and responder issues. I have other test scripts for other platforms,  PKIs and responders.

If you need a responder to test against, let me know out of band.

Regards
Nigel Tedeschi
[SCEP Message test menu For MS.bat.txt](https://github.com/certnanny/sscep/files/279407/SCEP.Message.test.menu.For.MS.bat.txt)

[sscep.c_#49.patch.txt](https://github.com/certnanny/sscep/files/279400/sscep.c_.49.patch.txt)

**Comment by mbartosch** [comment]
After reading the code for some time (wish I hadn't) I believe the patch is OK.

**Comment by tedescn** [comment]
Thanks Martin for this, your efforts are appreciated. 

 

I will ask Thorsten if we can revisit adding to Linux tools repo? Do you want us to keep you in the loop here?

 

Regards

Nigel

**Pull request merged**